File: archive/nodejs/muse/algorithms/sorting/BubbleSort.js
export const sort = (arr) => {
  let margin = 0;
  let unsorted = arr.length;

  while (unsorted > 1) {
    margin = 0;
    for (let i = 1; i < unsorted; i++) {
      if (arr[i - 1] > arr[i]) {
        [arr[i - 1], arr[i]] = [arr[i], arr[i - 1]];
        margin = i;
      }
    }
    unsorted = margin;
  }
};

File: archive/ruby/muse/algorithms/sorting/BubbleSort.rb
module BubbleSort
  def self.sort(arr)
    margin = 0
    unsorted = arr.length

    while unsorted > 1
      margin = 0
      1.upto(unsorted - 1) do |i|
        if arr[i - 1] > arr[i]
          arr[i - 1], arr[i] = arr[i], arr[i - 1]
          margin = i
        end
      end
      unsorted = margin
    end
  end
end
